src/Maths/Silk.NET.Maths/Simd/Simd.GetWithElement.cs

```diff
@@ -0,0 +1,91 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+#if INTRINSICS
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.Intrinsics;
+#if SSE
+using System.Runtime.Intrinsics.X86;
+#endif
+#if AdvSIMD
+using System.Runtime.Intrinsics.Arm;
+#endif
+
+namespace Silk.NET.Maths
+{
+    /// <summary>
+    /// Provides several hardware-accelerated methods
+    /// for 128-bit vectors
+    /// </summary>
+    public static unsafe partial class Simd128
+    {
+        // these two should ONLY be used in the constant-sized unrolled loops.
+        // unfortunately there is an issue otherwise. See https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAJI8ovLrl5hcAbho1iAZiakGAYRoBvGg11MUDACoxcGABQBKHXu3U99hgDNoDM+IwNeDALwM6MzwYAHgYANRgwDGhyUgAOIMcAGwhsDAA+FjUIbgwA3gBqfKs7B1KAOQgAeQAHNws5Et0AX2tdVv0GCpq3Hk9i+1tS3V5HVzMOdwtAtN8AegAqccn52fDI6LiE5NSMrJyp2dnguDgGbkh8AV6wPAx2oaZyAE5LBvsW6iagA
+        [MethodImpl(Scalar.MaxOpt)]
+        private static Vector128<T> WithElement<T>(Vector128<T> vector, int index, T value) where T : unmanaged
```

Comment:
If we keep the manually written versions of these methods can we validate `index` please? This method is actually unsafe otherwise, rather than just (ab)using unsafe.

Reply:
Then won't it destroy the purpose of the method? Kai's [comment](https://github.com/dotnet/Silk.NET/pull/665#discussion_r741953852). Actually, I personally would prefer to remove manually written methods, because it's not a good idea to have them in long-run perspective.

So can we decide one and forever here? @HurricanKai

Reply:
I'm fine with removing them. We should for sure raise an issue on dotnet/runtime then, cause while not a huge deal and only used in fallback paths it'd be nice to have that anyways

Reply:
I personally am not keen on not validating user input (segfaults are ugly), but can let this slide if Kai wants.